src/wallet/walletdb.cpp

```diff
@@ -647,9 +656,7 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             ssValue >> strValue;
             pwallet->LoadDestData(DecodeDestination(strAddress), strKey, strValue);
         } else if (strType == DBKeys::HDCHAIN) {
-            CHDChain chain;
-            ssValue >> chain;
-            pwallet->GetOrCreateLegacyScriptPubKeyMan()->LoadHDChain(chain);
+            if (!LoadHDChain(pwallet, ssKey, ssValue, strErr)) return false;
```

Comment:
Commit a0bc6bda272e7336deb55bdbf535534ed3a40306 currently doesn't compile, since the call to `LoadHDChain(...)` has too many arguments:
```suggestion
            if (!LoadHDChain(pwallet, ssValue)) return false;
```

Reply:
Fixed.